app/routes.py

```diff
@@ -1,2 +1,199 @@
+                "description": task.description,
+                "is_complete": False if not task.completed_at else True}}, 200)
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    if request.method == "PATCH":
+        task = Task.query.get(task_id)
+        if task is None:
+            return make_response("", 404)
+
+        task.completed_at = datetime.now()
+        db.session.commit()
+
+        slack_url = 'https://slack.com/api/chat.postMessage'
+        slack_tk = os.environ.get("SLACKBOT_TOKEN")
+
+        r = requests.post(slack_url,
+                          data=f"channel=task-notifications&text=Someone just completed the task {task.title}",
+                          headers={
+                              "Content-type": "application/x-www-form-urlencoded", "Authorization": f"{slack_tk}"}
+                          )
+        print(r.text)
+
+        return make_response({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if not task.completed_at else True}}, 200)
+
+# Add a goal and get all goals
+
+
+@goals_bp.route("", methods=["GET", "POST"])
+def handle_goals():
+    if request.method == "GET":
+        goals = Goal.query.all()
+        goals_response = []
+        for goal in goals:
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title,
+            })
+        return jsonify(goals_response)
+    elif request.method == "POST":
+        req_body = request.get_json()
+
+        # check for missing data
+        if "title" not in req_body:
+            return make_response({"details": "Invalid data"}, 400)
+        new_goal = Goal(title=req_body["title"])
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return make_response({"goal": {
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+        }}, 201)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return make_response("Item not found", 404)
+
+    if request.method == "GET":
+        return {"goal": {
+            "id": goal.goal_id,
+            "title": f"{goal.title}"
+        }}
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        if "title" not in form_data:
+            return make_response({"details": "Invalid data"}, 400)
+        goal.title = form_data["title"]
+        db.session.commit()
+
+        return make_response({"goal": {
+            "id": goal.goal_id,
+            "title": f"{goal.title}"
+        }})
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'})
```

Comment:
Go ahead and follow what you've been doing add the route here `@goal_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])` you want to get each Task to add it to the goal using the tasks field you created in the model.